src/lib/helper/buildPaginationHelper.ts

```diff
@@ -0,0 +1,162 @@
+    }
+
+    // Add element at middle left position if applicable
+    if (
+        numOfPages >= 5 &&
+    currentPageNumber >= 4 &&
+    numOfPages - currentPageNumber >= 1
+    ) {
+        pageItems.push(
+            createPageItem(currentPageNumber - 1, currentPageNumber, false, urlPrefix, searchQuery)
+        );
+    }
+
+    // Add element at middle position if applicable
+    if (
+        numOfPages >= 5 &&
+    currentPageNumber >= 3 &&
+    numOfPages - currentPageNumber >= 2
+    ) {
+        pageItems.push(
+            createPageItem(currentPageNumber, currentPageNumber, false, urlPrefix, searchQuery)
+        );
+    }
+
+    // Add element at middle right position if applicable
+    if (
+        numOfPages >= 5 &&
+    currentPageNumber >= 2 &&
+    numOfPages - currentPageNumber >= 3
+    ) {
+        pageItems.push(
+            createPageItem(currentPageNumber + 1, currentPageNumber, false, urlPrefix, searchQuery)
+        );
+    }
+
+    // Add second-to-last element if applicable - possible ellipsis
+    if (numOfPages >= 4) {
+        const isEllipsis = numOfPages >= 5 && numOfPages - currentPageNumber >= 4;
+        pageItems.push(
+            createPageItem(numOfPages - 1, currentPageNumber, isEllipsis, urlPrefix, searchQuery)
+        );
+    }
+
+    // Add last element if applicable
+    if (numOfPages > 1) {
+        pageItems.push(
+            createPageItem(numOfPages, currentPageNumber, false, urlPrefix, searchQuery)
+        );
+    }
+
+    pagination.items = pageItems;
+    return pagination;
+};
+
+const createPageItem = (
+    pageNumber: number,
+    currentPageNumber: number,
+    isEllipsis: boolean,
+    prefix: string,
+    searchQuery: string
+): PageItem => {
+    if (isEllipsis) {
+        return {
+            ellipsis: true
+        };
+    }
+    return {
+        current: currentPageNumber === pageNumber,
+        number: pageNumber,
+        href: `${prefix}?page=${pageNumber}${searchQuery}`
+    };
+};
+
+export const sortAndSearch = (
+    items: Association[],
+    search: string
+): Association[] | undefined => {
+    if (!items?.length) {
+        return;
+    }
+
+    // sort
+    items.sort((a, b) => {
+        if (a.companyName < b.companyName) {
+            return -1;
+        }
+        if (a.companyName > b.companyName) {
+            return 1;
+        }
+        return 0;
+    });
+
+    // filter / search
+    if (search && search.length) {
+        items = items.filter((item) => {
+            return (
+                item.companyName.includes(search.toUpperCase()) ||
+        item.companyNumber.includes(search.toUpperCase())
+            );
+        });
+    }
+
+    return items;
+};
+
+export const paginatedSection = (
+    items: Association[] | undefined,
+    page: number
+): Association[] | undefined => {
+    // paginated associations for display
+    if (!items?.length) {
+        return;
+    }
+    const startIndex = (page - 1) * ITEMS_PER_PAGE;
```

Comment:
Shouldn't the number of items per page come from the associations object? The data structure provides that information. 
![Screenshot 2024-03-20 at 15 18 11](https://github.com/companieshouse/your-companies-web/assets/150266545/36f2f5b2-ea1f-45f5-b697-9df369e17e4e)

Reply:
The function has been updated to get the value from the associations request.